File: brentversal/src/main/java/com/brentversal/agency/constant/ConsultationStatus.java
package com.brentversal.agency.constant;

// 상담 요청의 진행 상태를 나타내는 열거형 상수
//
// 상담이 진행되는 순서를 그대로 담았다.
//   상담 요청 -> 상담 확정 -> 상담 완료
// 중간에 진행하지 않기로 하면 어느 단계에서든 종료가 될 수 있다.
//
// 중개인 마이페이지의 "문의 관리" 필터(전체 / 상담 요청 / 상담 확정 / 상담 완료 / 종료)와 짝을 이룬다.
public enum ConsultationStatus {
    REQUESTED("상담 요청"),  // 사용자가 요청을 보냈고 중개인이 아직 답변하지 않은 상태
    ACCEPTED("상담 확정"),   // 중개인이 답변을 보내 일정이 잡힌 상태
    DONE("상담 완료"),       // 상담이 끝난 상태
    CLOSED("종료");          // 진행하지 않기로 했거나 취소된 상태

    // 화면에 그대로 보여 줄 한글 라벨.
    // 배지 색상 같은 표현은 서버가 아니라 프론트에서 정한다.
    private final String label;

    ConsultationStatus(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}

File: frentversal/src/utils/consultationStatus.ts
// 상담 상태를 화면에 표시할 때 쓰는 값들을 한곳에 모아 둔 파일
//
// 마이페이지·내 중개사무소·답변하기 세 화면이 같은 배지 색과 같은 필터를 쓰기 때문에,
// 각 화면에 흩어 두면 나중에 상태가 하나 늘었을 때 세 군데를 모두 고쳐야 한다.
//
// 한글 이름(statusLabel)은 서버가 내려주므로 여기서 정하지 않는다.
// 색상은 화면 표현이라 서버가 아니라 프론트에서 정한다.

import type { ConsultationStatus } from '../types/MyAgency';

// common.css 의 .status.green / .orange / .purple / .gray 와 짝을 이룬다
export const CONSULTATION_STATUS_COLORS: Record<ConsultationStatus, string> = {
    REQUESTED: 'orange', // 상담 요청 - 답변이 필요하므로 눈에 띄는 색
    ACCEPTED: 'purple',  // 상담 확정
    DONE: 'green',       // 상담 완료
    CLOSED: 'gray',      // 종료
};

// 문의 관리의 드롭다운 필터.
// value 는 서버의 ConsultationStatus 이름과 똑같아야 한다(ALL 만 예외로 '전체'를 뜻한다).
export const CONSULTATION_FILTERS = [
    { value: 'ALL', label: '전체' },
    { value: 'REQUESTED', label: '상담 요청' },
    { value: 'ACCEPTED', label: '상담 확정' },
    { value: 'DONE', label: '상담 완료' },
    { value: 'CLOSED', label: '종료' },
];
